runner/startup.sh

```diff
@@ -32,6 +32,10 @@ if [ -z "${RUNNER_NAME}" ]; then
   exit 1
 fi
 
+if [ -n "${DOCKER_DEFAULT_ADDRESS_POOL_BASE}" ] && [ -n "${DOCKER_DEFAULT_ADDRESS_POOL_SIZE}" ]; then
+  jq ".\"default-address-pools\" = [{\"base\": \"${DOCKER_DEFAULT_ADDRESS_POOL_BASE}\", \"size\": ${DOCKER_DEFAULT_ADDRESS_POOL_SIZE}}]" /etc/docker/daemon.json > /tmp/.daemon.json && /tmp/.daemon.json /etc/docker/daemon.json
+fi
```

Comment:
Note that this needs to be in https://github.com/actions-runner-controller/actions-runner-controller/blob/master/runner/entrypoint-dind-rootless.sh and https://github.com/actions-runner-controller/actions-runner-controller/blob/master/runner/entrypoint-dind.sh but https://github.com/actions-runner-controller/actions-runner-controller/blob/master/runner/entrypoint.sh

cc/ @toast-gear- this might answer your question in DM